keanu-python/tests/test_sampling.py

```diff
@@ -53,3 +58,46 @@ def test_down_sample_interval(net):
 
     expected_num_samples = draws / down_sample_interval
     assert all(len(vertex_samples) == expected_num_samples for vertex_id, vertex_samples in samples.items())
+
+
+@pytest.mark.parametrize("algo", [
+    ("metropolis"),
+    ("hamiltonian")
+])
+def test_can_iter_through_samples(algo, net):
+    draws = 10
+    samples = iterate_samples(net=net, sample_from=net.get_latent_vertices(), algo=algo, down_sample_interval=1)
+    count = 0
+    for sample in islice(samples, draws):
+        count += 1
+    assert count == draws
+
+
+@pytest.mark.parametrize("algo", [
+    ("metropolis"),
+    ("hamiltonian")
+])
+def test_iter_returns_same_result_as_sample(algo):
+    draws = 100
+    model = Thermometer.model()
+    net = BayesNet(model.temperature.get_connected_graph())
+    set_starting_state(model)
+    samples = sample(net=net, sample_from=net.get_latent_vertices(), algo=algo, draws=draws)
+    set_starting_state(model)
+    iter_samples = iterate_samples(net=net, sample_from=net.get_latent_vertices(), algo=algo)
+    iter_samples_dict = defaultdict(float)
+
+    for next_sample in islice(iter_samples, draws):
```

Comment:
pandas will have some out-the-box functionality to get the average over a data frame of samples

Reply:
any suggestions? It's iterating over data frames so I can't think of a better way than this

Reply:
Basically, dump all your vertex_samples into a dataframe and then call `mean()` on it.

	>>> import pandas as pd
	>>> df = pd.DataFrame([[1, 2], [10, 12]])
	>>> df.mean()
	0    5.5
	1    7.0
	dtype: float64